src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -967,7 +967,15 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
             IEnumerator list = LanguagePrimitives.GetEnumerator(lval);
             if (list == null)
             {
-                string lvalString = lval?.ToString() ?? string.Empty;
+                string lvalString;
+                if (ExperimentalFeature.IsEnabled("PSCultureInvariantReplaceOperator"))
+                {
+                    lvalString = PSObject.ToStringParser(context, lval) ?? string.Empty;
```

Comment:
`PSObject.ToStringParser` won't return a null value, so `?? string.Empty` can be removed.

Reply:
https://github.com/PowerShell/PowerShell/blob/384a7ba2d7b66b7099ac840915cbca36ae0494db/src/System.Management.Automation/engine/MshObject.cs#L1370

looks like a bug nonetheless it can return null.

Reply:
if `obj` is null, it won't reach that far, see
https://github.com/PowerShell/PowerShell/blob/384a7ba2d7b66b7099ac840915cbca36ae0494db/src/System.Management.Automation/engine/MshObject.cs#L1313-L1316

Reply:
ToString() can return null by design.

Reply:
Yes, but only if explicitly overridden to return null ..., which is practically not possible.
There will be too many places to change if we consider calling `ToString()` on a non-null object could return `null`.

Reply:
>  very much unlikely

The code process an generic/custom object so it is still possible.
If we enable nullable this will be reported.
We rarely use ToString() directly and if we enable nullable in the MshObject.cs we get dozen of errors for other cases. :-)

I added new commit.

Reply:
Some tests failed so I revert last commit. 
I think I will pull it in follow PR so that we could discuss.